src/VehicleSetup/JoystickConfigButtons.qml

```diff
@@ -163,24 +165,85 @@ ColumnLayout {
                     }
                 }
 
-                FactComboBox {
-                    id:             mainJSButtonActionCombo
-                    width:          ScreenTools.defaultFontPixelWidth * 15
-                    fact:           hasFirmwareSupport ? controller.getParameterFact(-1, parameterName) : null;
-                    visible:        hasFirmwareSupport
-                    indexModel:     false
+                QGCComboBox {
+                    id:                         buttonActionCombo
+                    width:                      ScreenTools.defaultFontPixelWidth * 26
+                    property Fact fact:         controller.parameterExists(-1, `BTN${index}_FUNCTION`) ? controller.getParameterFact(-1, `BTN${index}_FUNCTION`) : null
+                    property Fact fact_shift:   controller.parameterExists(-1, `BTN${index}_SFUNCTION`) ? controller.getParameterFact(-1, `BTN${index}_SFUNCTION`) : null
+                    property var factOptions:   fact ? fact.enumStrings : [];
+                    property var qgcActions:    _activeJoystick.assignableActionTitles.filter(function(s){return s.includes("Camera") || s.includes("Stream") || s.includes("Zoom") || s.includes("Gimbal")  || s.includes("No Action")})
```

Comment:
You could break this function in a new line to make it easier to read and maintain :laughing: 
```
property var qgcActions: _activeJoystick.assignableActionTitles.filter(function(string){
    return s.includes("Camera") \
        || s.includes("Stream") \
        || s.includes("Stream") \
        || s.includes("Zoom") \
        || s.includes("Gimbal") \
        || s.includes("No Action") \
    })
```

or

```
property var qgcActions: _activeJoystick.assignableActionTitles.filter(function(string){
    return [
        s.includes("Camera")
        , s.includes("Stream")
        , s.includes("Stream")
        , s.includes("Zoom")
        , s.includes("Gimbal")
        , s.includes("No Action")
    ].some(Boolean)})
```